src/gui/widgets/text_box_base.cpp

```diff
@@ -645,6 +679,14 @@ void text_box_base::signal_handler_sdl_key_down(const event::ui_event event,
 			handled = true;
 			break;
 
+		case SDLK_TAB:
+			if(!is_editable())
```

Comment:
This one seems a little questionable. The tab key could be used either to insert a tab or to switch focus to a different field, depending on the context. So exiting early in `text_box_base` seems wrong; `handle_key_tab` should probably check `is_editable` itself if it's planning to insert a tab.

Reply:
> handle_key_tab should probably check is_editable itself if it's planning to insert a tab.

So the check should be moved inside `handle_key_tab`?

Reply:
Yes, I think so, and only if `handle_key_tab` wants to insert a tab rather than moving focus to another field.

Reply:
Check moved inside handle_key_tab, so it is dealt with in the subclass.